app/models/user.rb

```diff
@@ -94,20 +96,12 @@ def school
     end.order(:updated_at, :id)
   }
 
-  scope :includes_school, lambda {
-    includes(
-      early_career_teacher_profile: %i[cohort school],
-      mentor_profile: %i[cohort school],
-      npq_profiles: :school,
-    )
+  scope :is_ecf_participant, lambda {
+    includes(participant_profile: :school_cohort).joins(:participant_profile).merge(ParticipantProfile.ecf.active)
```

Comment:
It scares me that `has_one :participant_profile` not `has_many :participant_profiles`

Reply:
When we move to `has_many` I imagine it will be a simple matter of switching the name to plural, right?

Am I missing something really scary?

Reply:
There are some implicit assumptions that there will only be one, I think

Reply:
Especially around display of NPQ/mentors

Reply:
I think the only place where we surface npqs and mentors at the same time is the admin panel, right?

I don't really know how we want to display them there, the obvious minimal implementation would display two copies of the same user, right?

Any more sophisticated solution is a bit out of my comfort zone and we should probably get a designers' opinion. Like, we could list the roles for each user, we could have a separate page for searching npq participants as opposed to ecf participants, we could probably have a few more ways of solving this problem. And I am not sure we know which one would be the most useful.

Reply:
No action here, just expressing fear